x-pack/plugins/observability_solution/entity_manager/server/lib/entities/transform/generate_metadata_aggregations.ts

```diff
@@ -34,7 +34,7 @@ export function generateLatestMetadataAggregations(definition: EntityDefinition)
   return definition.metadata.reduce(
     (aggs, metadata) => ({
       ...aggs,
-      [`entity.metadata.${metadata.destination}`]: {
+      [`entity.metadata.${metadata.destination ?? metadata.source}`]: {
```

Comment:
Maybe using https://zod.dev/?id=preprocess or https://zod.dev/?id=transform?

Reply:
can we do the same [transform operation](https://github.com/elastic/kibana/blob/main/x-pack/packages/kbn-entities-schema/src/schema/common.ts#L91) on the object, and assign `object.source` to `object.destination` if it's not set ?

Reply:
One challenge with this is that the type would still mark `destination` as optional. 
I'll try to handle it in preprocess which happens before validation and mark destination as required.

Reply:
When you use the `string` format, it does that:

```typescript
export const metadataSchema = z
  .object({
    source: z.string(),
    destination: z.optional(z.string()),
    limit: z.optional(z.number().default(1000)),
  })
  .or(z.string().transform((value) => ({ source: value, destination: value, limit: 1000 })));
```

How was this defined? `{ source: "foo", limit: 100 }`?

Reply:
If we change it to this:
```typescript
export const metadataSchema = z
  .object({
    source: z.string(),
    destination: z.optional(z.string()),
    limit: z.optional(z.number().default(1000)),
  })
  .transform((metadata) => ({
    ...metadata,
    desitination: metadata.destination ?? metadata.source,
    limit: metadata.limit ?? 1000,
  }))
  .or(z.string().transform((value) => ({ source: value, destination: value, limit: 1000 })));
```
then when you set the value to `{ "source": "host.name" }` it will parse it as `{ "source": "host.name", "destination": "host.name", "limit": 1000 }`